python/ray/dashboard/subprocesses/handle.py

```diff
@@ -204,206 +190,121 @@ async def _do_periodic_health_check(self):
-            request=request, response_fut=self.loop.create_future()
-        )
-        self.active_requests.put_new(request_id, new_active_request)
-        if request is None:
-            body = b""
-        else:
-            body = await request.read()
-        self._send_message(
-            RequestMessage(request_id=request_id, method_name=method_name, body=body)
-        )
-        return await new_active_request.response_fut
-
-    def _send_message(self, message: ChildBoundMessage):
-        self.child_bound_queue.put(message)
+        if resp_type == "http":
+            return await self.proxy_http(request)
+        if resp_type == "stream":
+            return await self.proxy_stream(request)
+        if resp_type == "websocket":
+            return await self.proxy_websocket(request)
+        raise ValueError(f"Unknown response type: {resp_type}")
+
+    async def proxy_http(
+        self, request: aiohttp.web.Request
+    ) -> aiohttp.web.StreamResponse:
+        """
+        Proxy handler for HTTP API
+        It forwards the method, query string, headers, and body to the backend.
+        """
+        url = f"http://localhost{request.path_qs}"
+        body = await request.read()
+
+        async with self.session.request(
+            request.method, url, data=body, headers=filter_headers(request.headers)
+        ) as resp:
+            resp_body = await resp.read()
+            return aiohttp.web.Response(
+                status=resp.status, headers=filter_headers(resp.headers), body=resp_body
+            )
 
-    @staticmethod
-    async def handle_stream_response_start(
-        request: aiohttp.web.Request, first_data: bytes
+    async def proxy_stream(
+        self, request: aiohttp.web.Request
     ) -> aiohttp.web.StreamResponse:
-        # TODO: error handling
-        response = aiohttp.web.StreamResponse()
-        response.content_type = "text/plain"
-        await response.prepare(request)
-        await response.write(first_data)
-        return response
-
-    @staticmethod
-    async def handle_stream_response_data(
-        prev_fut: Awaitable[aiohttp.web.StreamResponse], data: bytes
+        """
+        Proxy handler for streaming responses.
+        It forwards the method, query string, and body to the backend.
+        """
+        url = f"http://localhost{request.path_qs}"
+        body = await request.read()
+        async with self.session.request(
+            request.method, url, data=body, headers=filter_headers(request.headers)
+        ) as backend_resp:
+            client_resp = aiohttp.web.StreamResponse(status=backend_resp.status)
+            await client_resp.prepare(request)
+
+            async for chunk in backend_resp.content.iter_chunked(1024):
+                await client_resp.write(chunk)
+            await client_resp.write_eof()
+            return client_resp
+
+    async def proxy_websocket(
+        self, request: aiohttp.web.Request
     ) -> aiohttp.web.StreamResponse:
```

Comment:
```suggestion
    ) -> aiohttp.web.WebSocketResponse:
```

?

Reply:
Acutally this works because `WebSocketResponse` is a sub-class of `StreamResponse`. But I'll change this.